src/js/program/media-controller.js

```diff
@@ -121,7 +121,7 @@ export default class MediaController extends Eventable {
             playReason
         });
         // Immediately set player state to buffering if these conditions are met
-        if (video && !video.paused) {
+        if (video ? !video.paused : model.get(PLAYER_STATE) !== STATE_PAUSED) {
```

Comment:
It would be safer to check the for the one state that prevents impressions from firing.

You said in the description it's the play state, or is it the idle state? We should not accept both cases.

`video.paused` is a flag, not part of a state enum, so checking that the player is not paused is not the same as checking if video is expected to play after this point.